### ISSUE TITLE
Crash halfway on  itemCount 0 must be greater than or equal to 1

### ISSUE THREAD

**Issue by tomvantilburg** [Geodan/pg2b3dm#23]
Command:
```
docker run \
	-v $(pwd)/data/tiles:/app/output \
	-it geodan/pg2b3dm \
	-h metis -U tomt -d maquette_nl -p 5433 \
	-t tmp.betterroofs_extruded 
```
output:
```
tool: pg2b3dm 0.3.1.0
Start processing....
Calculating bounding boxes...
Writing tileset.json...
Writing 24 tiles...
Progress: tile 7 - 29.17%
Unhandled Exception: System.ArgumentOutOfRangeException: itemCount 0 must be greater than or equal to 1.
Parameter name: itemCount
   at SharpGLTF.Guard.MustBeGreaterThanOrEqualTo[TValue](TValue value, TValue min, String parameterName)
   at SharpGLTF.Schema2.Accessor.SetData(BufferView buffer, Int32 bufferByteOffset, Int32 itemCount, DimensionType dimensions, EncodingType encoding, Boolean normalized)
   at SharpGLTF.Schema2.Accessor.SetVertexData(BufferView buffer, Int32 bufferByteOffset, Int32 itemCount, DimensionType dimensions, EncodingType encoding, Boolean normalized)
   at SharpGLTF.Schema2.Accessor.SetVertexData(MemoryAccessor src)
   at SharpGLTF.Schema2.Schema2Toolkit.CreateVertexAccessor(ModelRoot root, MemoryAccessor memAccessor)
   at SharpGLTF.Schema2.Schema2Toolkit.WithVertexAccessor(MeshPrimitive primitive, MemoryAccessor memAccessor)
   at SharpGLTF.Schema2.Schema2Toolkit.WithVertexAccessors(MeshPrimitive primitive, IEnumerable`1 memAccessors)
   at SharpGLTF.Geometry.PackedPrimitiveBuilder`1.CopyToMesh(Mesh dstMesh, Func`2 materialEvaluator)
   at SharpGLTF.Geometry.PackedMeshBuilder`1.CreateSchema2Mesh(ModelRoot root, Func`2 materialEvaluator)
   at SharpGLTF.Schema2.Schema2Toolkit.CreateMeshes[TMaterial,TvP,TvM,TvS](ModelRoot root, Func`2 materialEvaluator, MeshBuilder`4[] meshBuilders)
   at pg2b3dm.Program.WriteB3dm(List`1 geomrecords, Int32 tile_id) in /src/pg2b3dm/Program.cs:line 171
   at System.Threading.QueueUserWorkItemCallbackDefaultContext.Execute()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
   at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
```

**Comment by bertt** [comment]
can reproduce this

**Comment by bertt** [comment]
fixed in new release 0.3.2 (https://github.com/Geodan/pg2b3dm/commit/922837cbdb536167416abadeab38147b50800661)

**Comment by tomvantilburg** [comment]
Works! :)
![image](https://user-images.githubusercontent.com/1998463/62144894-d6f94180-b2f2-11e9-97a2-37b46e55bff1.png)

**Comment by tomvantilburg** [comment]
Still happening in 0.4:

```
Unhandled Exception: System.ArgumentOutOfRangeException: itemCount 0 must be greater than or equal to 1.
Parameter name: itemCount
   at SharpGLTF.Guard.MustBeGreaterThanOrEqualTo[TValue](TValue value, TValue min, String parameterName)
   at SharpGLTF.Schema2.Accessor.SetData(BufferView buffer, Int32 bufferByteOffset, Int32 itemCount, DimensionType dimensions, EncodingType encoding, Boolean normalized)
   at SharpGLTF.Schema2.Accessor.SetVertexData(BufferView buffer, Int32 bufferByteOffset, Int32 itemCount, DimensionType dimensions, EncodingType encoding, Boolean normalized)
   at SharpGLTF.Schema2.Accessor.SetVertexData(MemoryAccessor src)
   at SharpGLTF.Schema2.Schema2Toolkit.CreateVertexAccessor(ModelRoot root, MemoryAccessor memAccessor)
   at SharpGLTF.Schema2.Schema2Toolkit.WithVertexAccessor(MeshPrimitive primitive, MemoryAccessor memAccessor)
   at SharpGLTF.Schema2.Schema2Toolkit.WithVertexAccessors(MeshPrimitive primitive, IEnumerable`1 memAccessors)
   at SharpGLTF.Geometry.PackedPrimitiveBuilder`1.CopyToMesh(Mesh dstMesh, Func`2 materialEvaluator)
   at SharpGLTF.Geometry.PackedMeshBuilder`1.CreateSchema2Mesh(ModelRoot root, Func`2 materialEvaluator)
   at SharpGLTF.Schema2.Schema2Toolkit.CreateMeshes[TMaterial,TvP,TvM,TvS](ModelRoot root, Func`2 materialEvaluator, MeshBuilder`4[] meshBuilders)
   at Wkb2Gltf.GlbCreator.GetGlb(TriangleCollection triangles) in /src/wkb2gltf.core/GlbCreator.cs:line 55
   at pg2b3dm.Program.WriteTiles(NpgsqlConnection conn, String geometryTable, String geometryColumn, Double[] translation, Node node, String outputPath, String colorColumn) in /src/pg2b3dm/Program.cs:line 83
   at pg2b3dm.Program.WriteTiles(NpgsqlConnection conn, String geometryTable, String geometryColumn, Double[] translation, Node node, String outputPath, String colorColumn) in /src/pg2b3dm/Program.cs:line 96
   at pg2b3dm.Program.<>c.<Main>b__2_0(Options o) in /src/pg2b3dm/Program.cs:line 71
   at CommandLine.ParserResultExtensions.WithParsed[T](ParserResult`1 result, Action`1 action)
   at pg2b3dm.Program.Main(String[] args) in /src/pg2b3dm/Program.cs:line 77
```

**Comment by tomvantilburg** [comment]
```
docker run \
        -v $(pwd)/data/tiles:/app/output \
        -it geodan/pg2b3dm \
        -h metis -U tomt -d maquette_nl -p 5433 \
        -t results.roofs_colored \
        -r color
```

**Comment by bertt** [comment]
changed the way to assign roof colors, should be fixed in 0.4.1 (https://github.com/Geodan/pg2b3dm/commit/e406b47f4546c13bda60cf7fa72ae4cd6a7482d8)

**Comment by tomvantilburg** [comment]
Works!
